src/vsr/replica.zig

```diff
@@ -93,6 +94,10 @@ pub fn ReplicaType(
         const Journal = vsr.Journal(Self, Storage);
         const Clock = vsr.Clock(Time);
 
+        /// We use this allocator during open/init and then disable it.
+        /// An accidental dynamic allocation after open/init will cause an assertion failure.
+        revocable_allocator: RevocableAllocator,
+
```

Comment:
We should follow Zig's _unmanaged_ idiom here, by rather not storing the allocator on the replica at all. Typically, a struct will only store an allocator, if the struct is allowed to allocate after `init()`. This also saves memory in general for small structs like hash maps.

The allocator should rather be passed explicitly into `init()` and `deinit()`.

Reply:
That's what I did originally and in https://github.com/tigerbeetledb/tigerbeetle/pull/201#issuecomment-1279387782 you asked me to change it. 

I could pull the call the state change out of replica entirely, but then we'd have to duplicate the code in both main and the vopr and any other tests, and we'd also have to make the assert slightly looser by allowing allocation during the forest open at the end of Replica.open.

Reply:
I think there was a misunderstanding w.r.t. to that comment. :p